plugin/pkg/scheduler/algorithm/predicates/predicates_test.go

```diff
@@ -575,7 +575,188 @@ func TestPodFitsSelector(t *testing.T) {
 			fits: false,
 			test: "node labels are subset",
 		},
+		{
+			pod: &api.Pod{
+				ObjectMeta: api.ObjectMeta{
+					Annotations: map[string]string{
+						api.AffinityAnnotationKey: `
+						{"nodeAffinity": { "requiredDuringSchedulingIgnoredDuringExecution": {
+							"nodeSelectorTerms": [{
+								"matchExpressions": [{
+									"key": "foo",
+									"operator": "In",
+									"values": ["bar", "value2"]
+								}]
+							}]
+						}}}`,
+					},
+				},
+			},
+			labels: map[string]string{
+				"foo": "bar",
+			},
+			fits: true,
+			test: "required node affinity scheduling requirements matches node",
+		},
+		{
+			pod: &api.Pod{
+				ObjectMeta: api.ObjectMeta{
+					Annotations: map[string]string{
+						api.AffinityAnnotationKey: `
+						{"nodeAffinity": {
+							"requiredDuringSchedulingRequiredDuringExecution": {
+								"nodeSelectorTerms": [{
+									"matchExpressions": [{
+										"key": "foo",
+										"operator": "In",
+										"values": ["bar", "value2"]
+									}]
+								}]
+							},
+							"requiredDuringSchedulingIgnoredDuringExecution": {
+								"nodeSelectorTerms": [{
+									"matchExpressions": [{
+										"key": "foo",
+										"operator": "NotIn",
+										"values": ["bar", "value2"]
+									}]
+								}]
+							}
+						}}`,
+					},
+				},
+			},
+			labels: map[string]string{
+				"foo": "bar",
+			},
+			fits: true,
```

Comment:
this should be false; see my earlier comment about how to handle the situation where both are set